google/cloud/bigquery/client.py

```diff
@@ -278,10 +285,16 @@ def list_projects(
                 Iterator of :class:`~google.cloud.bigquery.client.Project`
                 accessible to the current client.
         """
+        path = "/projects"
+        span_attributes = {"path": path}
+        with create_span(
+            name="BigQuery.listProjects", attributes=span_attributes, client=self
+        ):
+            api_request = functools.partial(self._call_api, retry, timeout=timeout)
```

Comment:
This actually doesn't make any API calls. `functools.partial` creates a function object that will call `_call_api` with the specified arguments.

If you want `HTTPIterator` to actually create spans, you'll have to create a function that does this manually. Something like:

```python
def list_projects(...):
    ...
    def api_request(*args, **kwargs):
        with create_span(
            name="BigQuery.listProjects", attributes=span_attributes, client=self
        ):
            return self._call_api(retry, *args, timeout=timeout, **kwargs)
    return page_iterator.HTTPIterator(...)
```

Reply:
Acknowledged in a separate PR as per our conversation. Draft can be found here:
https://github.com/googleapis/python-bigquery/pull/228

Reply:
Since these spans aren't actually wrapping the "real" request, could you remove these changes in this PR? I don't want someone to think that' telemetry is actually implemented for list calls.

Reply:
Sure